src/cert_store.rs

```diff
@@ -150,6 +171,46 @@ impl CertStore {
         }
     }
 
+    /// Deletes a specified store within the context of the current user.
+    /// 
+    /// Deleting stores is not natural and should only be done if necessary
+    pub fn delete_current_user_store(which: &str) -> () {
+        unsafe {
+            let data = OsStr::new(which)
+                .encode_wide()
+                .chain(Some(0))
+                .collect::<Vec<_>>();
+            let _store = Cryptography::CertOpenStore(
+                Cryptography::CERT_STORE_PROV_SYSTEM_W,
+                Cryptography::CERT_QUERY_ENCODING_TYPE::default(),
+                Cryptography::HCRYPTPROV_LEGACY::default(),
+                Cryptography::CERT_SYSTEM_STORE_CURRENT_USER_ID
+                    << Cryptography::CERT_SYSTEM_STORE_LOCATION_SHIFT | Cryptography::CERT_STORE_DELETE_FLAG,
+                data.as_ptr() as *mut _,
+            );
+            if io::Error::last_os_error().raw_os_error() != Some(0) {
+                panic!("There was an error deleting the CertStore");
+            }
+        }
+    }
+
+    /// Deletes a cert as well as the cert's coresponding private key
+    /// 
+    /// This function will return an error if the provided cert does not have a private key
+    pub fn delete_cert_and_key(mut cert_context: CertContext) -> Result<(), DeleteCertAndKeyError> {
+        if cert_context.private_key().silent(true).compare_key(true).acquire().is_ok() {
+            if let Err(_err) = cert_context.del_key_container() {
+                return Err(DeleteCertAndKeyError::DeleteKeyContainerError);
```

Comment:
>DeleteCertAndKeyError::DeleteKeyContainerErro [](http://example.com/codeflow?start=27&length=45)

Don't you want to return the error returned by del_key_container.

Same applies to .delete()